src/test/java/co/rsk/federate/FederationWatcherTest.java

```diff
@@ -22,354 +22,342 @@
+            @Override
+            public void onActiveFederationChange(Optional<Federation> oldFederation, Federation newFederation) {
+                activeCalls.incrementAndGet();
+            }
 
-                @Override
-                public void onRetiringFederationChange(Optional<Federation> oldFederation, Optional<Federation> newFederation) {
-                    logger.retiringCalls++;
-                }
-            });
-        }
+            @Override
+            public void onRetiringFederationChange(Optional<Federation> oldFederation, Optional<Federation> newFederation) {
+                retiringCalls.incrementAndGet();
+            }
+        });
 
+        // Act
         rskListener.onBestBlock(null, null);
-        assertEquals(0, logger.activeCalls);
-        assertEquals(0, logger.retiringCalls);
-        verify(federationProvider, times(1)).getActiveFederationAddress();
-        verify(federationProvider, times(1)).getRetiringFederationAddress();
+
+        // Assert
+        assertEquals(0, activeCalls.get());
+        assertEquals(0, retiringCalls.get());
+        verify(federationProvider).getActiveFederationAddress();
+        verify(federationProvider).getRetiringFederationAddress();
         verify(federationProvider, never()).getActiveFederation();
         verify(federationProvider, never()).getRetiringFederation();
     }
 
     @Test
-    void doesntTriggerActiveOrRetiringFederationChange_noChange() throws Exception {
-        EthereumListenerAdapter rskListener = setupAndGetRskListener(Optional.of(federation1), Optional.of(federation2));
-        class EventsLogger
-        {
-            public int activeCalls = 0;
-            public int retiringCalls = 0;
-        }
-
-        when(federationProvider.getActiveFederationAddress()).thenReturn(federation1.getAddress());
-        when(federationProvider.getRetiringFederationAddress()).thenReturn(Optional.of(federation2.getAddress()));
-
-        EventsLogger logger = new EventsLogger();
-
-        for (int i = 0; i < 2; i++) {
-            watcher.addListener(new FederationWatcher.Listener() {
-                @Override
-                public void onActiveFederationChange(Optional<Federation> oldFederation, Federation newFederation) {
-                    logger.activeCalls++;
-                }
-
-                @Override
-                public void onRetiringFederationChange(Optional<Federation> oldFederation, Optional<Federation> newFederation) {
-                    logger.retiringCalls++;
-                }
-            });
-        }
-
+    void whenNoActiveAndRetiringChangeInFederation_shouldNotTriggerActiveOrRetiringFederationChange() throws Exception {
+        // Arrange
+        var rskListener = setupAndGetRskListener(Optional.of(FIRST_FEDERATION), Optional.of(SECOND_FEDERATION));
+        var activeCalls = new AtomicInteger(0);
+        var retiringCalls = new AtomicInteger(0);
```

Comment:
wdyt about a global setUp to reset these values?

Reply:
Seems to overcomplicate things for a local counter variable